src/components/panel/panel.js

```diff
@@ -938,6 +958,35 @@ MdPanelRef.prototype._configureClickOutsideToClose = function() {
 };
 
 
+/**
+ * Setup the focus traps. These traps will wrap focus when tabbing past the
+ * panel. When shift-tabbing, the focus will stick in place.
+ * @private
+ */
+MdPanelRef.prototype._configureTrapFocus = function() {
+  this._panelEl.attr('tabIndex', '-1');
```

Comment:
In truth, I have no clue why it works with this line and doesn't work without it. This was code specified in the focus traps of dialog.